Develop

vladbarsukov opened the pull request:
Ссылка на предыдущий pull request
https://github.com/vladbarsukov/mesto-project/pull/7
<img width="804" alt="image" src="https://user-images.githubusercontent.com/101993865/198696513-cfd61542-5231-47c3-9424-2e679b5d17a4.png">

vladbarsukov committed:
исправил ошибку разблокировки кнопки

vladbarsukov committed:
исправил card, userInfo

shumkova committed:
правит FormValidator и Попапы

vladbarsukov committed:
исправил оверлей попапа с картинкой

vladbarsukov committed:
исправил секшн

shumkova committed:
выносит api из Card, изменяет renderer для Section

vladko-89 commented on src/index.js:
```diff
@@ -89,13 +143,9 @@ profileAddButton.addEventListener("mousedown", (evt) => {
-      jobInput.value = userData.about;
-    })
-
+  let user = userInfo.getUserInfo();
```
Надо исправить:

Переменная не перезаписывается. Значит нужно объявить ее через `const`

vladko-89 commented on src/index.js:
```diff
@@ -28,53 +28,107 @@ const formEditProfileValidator = new FormValidator(validationSettings, formEditP
-    .finally(() => {
-      popupAddCard.submitButton.textContent = "Создать";
+const cardList = new Section({
+  renderer: (cardData, userId) => {
```
Надо исправить:

Пробрасывать `userId` через класс Section не лучшее решение. Старайтесь подходить к разработке так, чтобы не передавать лишние данные через классы в которых эти данные не применяются и не используются. Предлагаю сохранить id в `UserInfo` или же в глобальную переменную и передавать ее непосредственно в `Card` минуя проброс через `Section`.

vladko-89 commented on components/PopupWithForm.js:
```diff
@@ -1,22 +1,26 @@
 
   _getInputValues() {
-    return this._form.elements;
+    const formElements = Array.from(this._form.elements);
```
Надо исправить:

Коллекция полей ввода формы статична. Нужно занести ее в конструктор при создании экземпляра и использовать в методе, а не искать каждый раз в DOM

vladko-89 commented on src/index.js:
```diff
@@ -28,53 +28,107 @@ const formEditProfileValidator = new FormValidator(validationSettings, formEditP
+        card._deleteButton.disabled = true;
+        return api.deleteCardFromServer(cardId)
+          .then(() => {
+            card._element.remove();
```
Методы и поля в которых встречается `_` являются защищенными и не должны использоваться вне тела класса

vladko-89 reviewed (commented):
Добрый день! Вы отлично потрудились и многое удалось исправить. Но еще остались места над чем нужно поработать:

- Все поля и методы класса с `_` являются защищенными и не должны использоваться вне тела класса. Используйте методы удаления класса карточки вместо remove
- Пробрасывать `userId` через класс Section не очень хорошо. В комментарии предложил альтернативный вариант.
- Использовать `let` при объявлении переменных нужно только если переменная перезапишется. Подсветил место комментарием
- Коллекция полей ввода в PWF статична. Вынесите ее в конструктор и переиспользуйте.

Исправьте замечания и работа будет принята

vladbarsukov committed:
сделал константой переменную юзер

присвоил userId переменной и пробросил напрямую в card, занес коллекцию полей ввода форм в конструктор

vladbarsukov committed:
Исправил методы класса Card

shumkova committed:
упрощает переключатель кнопки удаления карточки

vladko-89 commented on components/PopupWithForm.js:
```diff
@@ -1,22 +1,25 @@
     this._form = this._element.querySelector(".popup__form");
-    this.submitButton = this._form.querySelector(".popup__submit");
+    this._formElements = Array.from(this._form.elements);
+    this._formInputValues = {};
```
Надо исправить:

Объект как раз нужно каждый раз пересоздавать в методе. Если к примеру добавить карточку, а затем поменять данные пользователя или аватар — объект на втором этапе будет содержать в себе данные карточки и дополнит их ключами и значениями из второй формы, получается что второй раз будут отправлены лишние данные, такого допускать не следует

vladko-89 reviewed (commented):
Добрый день! Отлично, вы почти все исправили, остался один момент:

- Объект который возвращает метод `_getInputValues` нужно пересоздавать в метода, а не записывать в конструктор. Причину расписал в комментарии

Исправьте это замечание и работа будет принята

vladbarsukov committed:
исправвил _formInputValues

vladko-89 reviewed (commented):
Вот теперь все отлично! Работа принята

Желаю успеха в дальнейшем обучении!
